### PR TITLE
[examples] Add CPP multiplex example

### PR DESCRIPTION
**Pull request by eugeneo**

### CODE REVIEW

File: examples/cpp/multiplex/BUILD [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 the gRPC authors.
```
Comment (yashykt) [inline examples/cpp/multiplex/BUILD:1]: fix copyright year

File: examples/cpp/multiplex/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# Copyright 2018 gRPC authors.
```
Comment (yashykt) [inline examples/cpp/multiplex/CMakeLists.txt:1]: copyright year here as well

**Review by yashykt — APPROVED** [review]

File: examples/cpp/multiplex/CMakeLists.txt (hunk 2 above)
Comment (eugeneo) [reply examples/cpp/multiplex/CMakeLists.txt]: Done

File: examples/cpp/multiplex/BUILD (hunk 1 above)
Comment (eugeneo) [reply examples/cpp/multiplex/BUILD]: Done

**Review by eugeneo — COMMENTED** [review]
Thank you for the review!

**Pull request merged**